src/__flow-tests__/nav-test.js

```diff
@@ -64,11 +64,11 @@ function testNavigatorTypes() {
     <Stack.Navigator>
       {/* Happy case is happy */}
       <Stack.Screen name="Profile" component={Profile} />
-      {/* $FlowExpectedError[incompatible-type] - mismatch of name with route prop */}
+      {/* FAIL - Should error but doesn't! on mismatch of name with route prop */}
       <Stack.Screen name="Profile1" component={Profile12} />
       {/* FAIL - Should error but doesn't! on mismatch of name with navigation prop */}
       <Stack.Screen name="Profile2" component={Profile12} />
-      {/* $FlowExpectedError[incompatible-type] - name that isn't even in the list */}
+      {/* $FlowExpectedError[prop-missing] - name that isn't even in the list */}
```

Comment:
> We also lose type-checking on the part of a screen's interface that says which navigator(s) it can go on.

I think we said on the call today that this is unfortunate but tolerable. Fixing the TypeScript types would take more time and effort than we can afford.

Reply:
> Separately, I've just noticed that Flow doesn't complain if I make this change, either at the tip of the branch, or on `main`:
> …
> It sure seems like it should, since 'asdfjkl;' isn't a subtype of `$Keys<AppNavigatorParamList>`. But I think that's something else/unrelated.

Yeah. That's this failing type-test in main:
```js
      {/* Should error but doesn't! on mismatch of name with navigation prop */}
      <Stack.Screen name="Profile2" component={Profile12} />
```

Or as it's worded at the tip of the PR, as discussed above:
```js
      {/* FAIL - Should error but doesn't! on mismatch of name with navigation prop */}
      <Stack.Screen name="Profile2" component={Profile12} />
```



> (However, it was a bad one that would've baffled me if I didn't already know I was misusing `PasswordAuthScreen`.)

Ha, good to know. So I guess even the type-test that had been passing isn't that big of a regression, in practice.

Reply:
> Yeah. That's this failing type-test in main:

Hmm, really? In my example, I wrote something that isn't a valid type at all: `AppNavigationProp<'asdfjkl;'>`. I don't see that happening in the "Should error but doesn't!" example.

Reply:
Mmm, I see. Yeah, that's puzzling. I do reproduce (on main); and I don't know why Flow doesn't report an error there, given the bound on the parameter of `AppNavigationProp`:
```js
export type AppNavigationProp<
  +RouteName: $Keys<AppNavigatorParamList> = $Keys<AppNavigatorParamList>,
> = StackNavigationProp<GlobalParamList, RouteName>;
```